src/main/scala/chisel3/tester/ChiselTester.scala

```diff
@@ -0,0 +1,66 @@
+// See LICENSE for license details.
+
+package chisel3.tester
+
+import chisel3.experimental.MultiIOModule
+import firrtl.ExecutionOptionsManager
+import org.scalatest._
+import org.scalatest.exceptions.TestFailedException
+
+import scala.collection.mutable
+import scala.util.DynamicVariable
+
+/**
+  * Used to run simple tests that do not require a scalatest environment in order to run
+  * @param testName This will be used to generate a working directory in ./test_run_dir
+  */
+class ChiselTester private (testName: String) extends Assertions with ChiselTesterHelper with TestEnvInterface {
+  // Provide test fixture data as part of 'global' context during test runs
+
+  val topFileName = Some(testName)
+
+  private def runTest[T <: MultiIOModule](tester: BackendInstance[T])(testFn: T => Unit) {
+    batchedFailures.clear()
+
+    Context.run(tester, this, testFn)
+  }
+
+  def getTestOptions: TesterOptions = {
+    TesterOptions(topFileName.get, writeVcd = false)
+  }
+
+  // This should be the only user-called function
+  def test[T <: MultiIOModule](dutGen: => T)(testFn: T => Unit) {
+    runTest(Context.createDefaultTester(() => dutGen, getTestOptions, None))(testFn)
+  }
+
+  def test[T <: MultiIOModule](dutGen: => T, execOptions: ExecutionOptionsManager)(testFn: T => Unit) {
+    runTest(Context.createDefaultTester(() => dutGen, getTestOptions, Some(execOptions)))(testFn)
+  }
+}
+
+/**
+  * This is a simple tester that does not require that it be within the scope of a scalatest
+  * in order to run. This form is suitable for running in the Jupyter notebook.
+  *
+  * @todo When Phases, Stages is implemented add ability to change testing options.
+  */
+object ChiselTester {
```

Comment:
I think I'd prefer it for users to just import a specific package (eg `chisel.tester.main` - but a better name) and write `test(dutGen) { testFn }`. The test would have no name - the name would either be generated from the DUT name, or just 'anonymous'. In the future, we can consider extensions like `test(dutGen).withName(...) { testFn }`, but I don't see a pressing need for that in the immediate case.

This primarily means renaming this object and putting it in a subpackage that isn't visible if you just import testers (since it could conflict with another test environment interface). ChiselTester as a class can stay as-is, since users shouldn't interact with that directly. It might help if it had a more useful name, eg BareTester, that differentiates it from the other test environment interfaces.

Reply:
I agree this sounds cleaner, but the hidden consequence is tests overwriting each other possibly picking up bad artifacts or instead making more and more new directories for every test. This maybe ok here, but it should be pondered. In the case of MyBinder the latter is probably fine as the sessions go away. What's you call on the default way of generating a target dir name

Reply:
So I don't expect people to use this in production testing, because if you're running lots of tests you should probably be using ScalaTest or some similar powerful test framework. This is mainly for people who either don't want to set up the ScalaTest infrastructure (so probably pretty small-scale testing) or where ScalaTest isn't available (bootcamp).

I've usually considered temporary directories for testing as generated on a best-effort basis without guarantees. I'd actually prefer no artifacts were generated unless specifically requested by the user (for example, in debug mode), since disk IO takes time and the vast majority of the time won't be used. Possibly unavoidable with Verilator, but possible in treadle.

Under that model, where temporaries are needed for compilation, I think inferring from the DUT name is fine, or even just using the same directory that gets cleared and written for each test.

Reply:
Once again, can we rename this to something more descriptive than ChiselTester that specifically calls out the distinguishing characteristics?